### PR TITLE
[WIP] Error message for writing empty dataframe to file

### PR DESCRIPTION
**Pull request by cr458**
Related to #657

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Thanks! Can you add a test for it?

**Comment by cr458** [comment]
@jorisvandenbossche sure! Just thought I'd open up the PR in case someone else started work on it

**Comment by cr458** [comment]
@jorisvandenbossche added a test into `test_geodataframe.py`, seems like that should be the right place. I could clean this test file up in this PR as well, instead of opening a new one if you'd like?

File: geopandas/tests/test_geodataframe.py [hunk 1]
```diff
@@ -361,6 +361,14 @@ def test_mixed_types_to_file(self):
         with pytest.raises(ValueError):
             s.to_file(tempfilename)
 
+    def test_empty_to_file(self):
+        input_empty_df = GeoDataFrame()
+        tempfilename = os.path.join(self.tempdir, 'test.shp')
+        with pytest.raises(
+            ValueError, message="Cannot write empty DataFrame to file."):
```
Comment (jorisvandenbossche) [inline geopandas/tests/test_geodataframe.py]: to test the correct message, I think you need to use the `match=` keyword (the `message` keyword is to give a custom message if the *assert* failes, somewhat confusing I have to say ..)

Comment (cr458) [reply geopandas/tests/test_geodataframe.py]: Good catch. Pytest is really powerful but a little confusing at times :)

**Comment by jorisvandenbossche** [comment]
Thanks a lot!

**Comment by akrherz** [comment]
Lazy peanut gallery here wondering about this code change...  Why is it invalid to dump an empty GeoDataFrame to file whilst using the `GeoJSON` driver?  Isn't the expected result in that case:

```
{"type": "FeatureCollection", "features": []}
```

**Comment by martinfleis** [comment]
@akrherz It would probably be okay with GeoJSON, but that'd require driver-specific behaviour. We tend to leave driver-specificity to fiona if possible.

**Comment by joooeey** [comment]
@akrherz @martinfleis Then why not convert the error into a warning? It's a lot of work to special-case the empty case when working with large amounts of data.

**Comment by martinfleis** [comment]
@joooeey with the current IO implementation, we're not able to write it. If we changed this error into a warning, some other, less understandable, error would bubble up.

**Comment by joooeey** [comment]
@martinfleis That depends on the output format. With GeoJSON, if I comment out the check, no error occurs, and a GeoJSON representing an empty feature collection is written.

**Comment by martinfleis** [comment]
@joooeey Can you open a new issue with a suggestion to skip this check for GeoJSON? This discussion will get lost here.

**Pull request merged**